### PR TITLE
ADD: Pothole Detection using YOLOv8_Computer Vision (#1873)

### LINKED ISSUE #1873: 💡[FEATURE]: Add Pothole Detection using YOLOv8
**Is your feature request related to a problem? Please describe.**
There is no project in the repository that demonstrates road defect detection using object detection models. Pothole detection is a highly relevant real-world problem, especially in the Indian context, and learners have no reference for applying YOLO-based object detection to infrastructure safety — a growing area in both industry and research.

**Describe the solution you'd like**
Addition of a new project called "Pothole Detection using YOLOv8" under Computer Vision. The project will fine-tune a YOLOv8 model on a labeled pothole dataset to detect potholes in real-time from images or dashcam video. It will include bounding box visualization and confidence scores, implemented as a well-documented .ipynb notebook with a README.md.

Tech Stack: Python, YOLOv8 (Ultralytics), OpenCV, NumPy, Matplotlib
Dataset: Pothole Detection Dataset — available on Roboflow / Kaggle

**Describe alternatives you've considered**
A classical image processing approach using edge detection was considered, but YOLOv8 offers significantly better accuracy, real-time performance, and is the industry-standard approach for object detection tasks of this nature.

**Additional context**
I want to work on this under GSSoC '26
---
name: 💡 Feature Request
about: Suggest an interesting feature idea for this project
title: '💡[FEATURE]: '
labels: 'enhancement'
assignees: ''

### PR DESCRIPTION
**Pull request by vanshikaMaheshwari**
Closes #1873 
--
What I added
--

New folder Pothole_Detection_YOLOv8/ under the repository root
Complete Jupyter notebook with dataset download, EDA, YOLOv8 fine-tuning, evaluation metrics, and bounding box visualization on test images
Beginner-friendly README.md with setup instructions, results table, and project structure

Models Implemented
--

YOLOv8n (nano) — fine-tuned on Roboflow Pothole Detection Dataset

Evaluation Metrics
--
mAP@50
Precision
Recall
Training & validation loss curves

Highlights
--
First road defect / infrastructure safety project under Computer Vision in ML-CaPsule
Includes optional dashcam video inference block
Uses Roboflow API for clean dataset download directly inside the notebook
Highly relevant to the Indian road safety context
Dataset sourced from: [Roboflow Universe — Pothole Detection](https://universe.roboflow.com/roboflow-100/pothole-detection-dataset)

Checklist
--
[] I have read the contributing guidelines
[] Follows project folder structure
[] Beginner-friendly README included
[] Assigned issue #[ISSUE_NUMBER]
[] Contributing under GSSoC '26

@Niketkumardheeryan please review 🙌

### CODE REVIEW

**Comment by vanshikaMaheshwari** [comment]
Please review this PR @Niketkumardheeryan. Let me know if any change is required.

**Comment by vanshikaMaheshwari** [comment]
Please review this PR @Niketkumardheeryan . Let me know if any change is required.

File: Pothole_Detection/Pothole_Detection_YOLOv8.ipynb
Comment (Sibam-Paul) [inline Pothole_Detection/Pothole_Detection_YOLOv8.ipynb:1]: please make it notebook valid 

Image


can't see the file 
and u have to run the entire shell
if u have a dataset upload it in gdrive and fetch the link in here and rerun the cells

File: Pothole_Detection/README.md
Comment (Sibam-Paul) [inline Pothole_Detection/README.md:1]: follow this Pr #1952 format

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]

**Comment by Sibam-Paul** [comment]
we want models like not the basic ones for the better of this repo 
if u can add more features which is uncommon of something like which is rare it will be good 
and I will award u for that @vanshikaMaheshwari

**Comment by vanshikaMaheshwari** [comment]
Hi @Sibam-Paul, I've made all the requested changes — notebook is now valid JSON with all cells executed, dataset is fetched via gdown from Google Drive (no API key needed), and README follows PR #1952 format. Please review!

**Review by Sibam-Paul — APPROVED** [review]

**Comment by Sibam-Paul** [comment]
thanks for contribution
